GeeksCoreLibrary/Modules/PostalServices/PostNL/Services/PostNLService.cs

```diff
@@ -0,0 +1,194 @@
+                throw;
+            }
+            finally
+            {
+                await CleanLogs();
+
+                databaseConnection.ClearParameters();
+                databaseConnection.AddParameter("order_id", orderId);
+                databaseConnection.AddParameter("request", requestString);
+                databaseConnection.AddParameter("response", responseString);
+                databaseConnection.AddParameter("exception", exceptionMessage);
+
+                await databaseConnection.InsertOrUpdateRecordBasedOnParametersAsync("cust_postnl_log", 0UL);
+            }
+        }
+
+        /// <summary>
+        /// Generates a new label using the PostNL api
+        /// </summary>
+        /// <param name="orderId">The orderId of the order the label must be created for</param>
+        /// <param name="request">Request model containing all the data used for creating the label</param>
+        /// <returns>Model containing the information of the generated label</returns>
+        public async Task<ShipmentResponseModel> CreateTrackTraceLabel(string orderId, ShipmentRequestModel request)
+        {
+            var exceptionMessage = "";
+            var responseString = "";
+            var requestString = "";
+
+            try
+            {
+                var restClient = new RestClient(gclSettings.PostNlApiBaseUrl);
+                var restRequest = new RestRequest("/v1/shipment?confirm=true", Method.POST);
+                requestString = JsonConvert.SerializeObject(request, Formatting.None, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+                restRequest.AddParameter("application/json", requestString, ParameterType.RequestBody);
+                restRequest.AddJsonBody(request);
+                restRequest.AddHeader("apiKey", gclSettings.PostNlShippingApiKey);
+
+                var response = await restClient.ExecuteAsync<ShipmentResponseModel>(restRequest);
+
+                responseString = response.Content;
+                if (response.ErrorException != null)
+                {
+                    exceptionMessage = response.ErrorException.ToString();
+                }
+
+                return response.Data;
+            }
+            catch (Exception exception)
+            {
+                exceptionMessage = exception.ToString();
+                throw;
+            }
+            finally
+            {
+                await CleanLogs();
+
+                databaseConnection.ClearParameters();
+                databaseConnection.AddParameter("order_id", orderId);
+                databaseConnection.AddParameter("request", requestString);
+                databaseConnection.AddParameter("response", responseString);
+                databaseConnection.AddParameter("exception", exceptionMessage);
+
+                await databaseConnection.InsertOrUpdateRecordBasedOnParametersAsync("cust_postnl_log", 0UL);
```

Comment:
In de GCL willen we geen tabelnamen direct in query's hebben staan. Kun je hier een constante van maken en dan die constante in de query zetten?